src/Context/AnnotatedClassDiscovery.php

```diff
@@ -11,7 +11,13 @@
 use \Drupal\Core\Plugin\Discovery\AnnotatedClassDiscovery as CoreAnnotatedClassDiscovery;
 
 /**
- * Extends the annotation class discovery for usage with Rules context.
+ * Extends the annotation class discovery for usage with our modified annotations.
+ *
+ * We modify the annotations classes for ContextDefinition and for Condition.
+ * This class makes sure that our plugin managers apply our versions of those
```

Comment:
"of those Core" is confusing, let's just remove that part and say it's about "our own versions."